modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/DynamicInlineScroll.es.js

```diff
@@ -159,8 +159,18 @@ class DynamicInlineScroll extends PortletBase {
 	onScroll_(event) {
 		const {cur, initialPages, pages} = this;
 		const {target} = event;
-		let pageIndex = this.getNumber_(target.dataset.pageIndex);
-		let pageIndexMax = this.getNumber_(target.dataset.maxIndex);
+
+		let targetNode = target;
+		while (targetNode.parentNode && targetNode.nodeName !== 'UL') {
+			targetNode = targetNode.parentNode;
+		}
+
+		if (targetNode.nodeName !== 'UL') {
+			return;
+		}
+
+		let pageIndex = this.getNumber_(targetNode.dataset.pageIndex);
+		let pageIndexMax = this.getNumber_(targetNode.dataset.maxIndex);
```

Comment:
We can simplify search for `ul`. Instead of loop, we can use `closest`, something like:
```js
const listElement = target.closest('ul');
```
There is probably no need for check if `ul` exists, but it's not a problem to leave it in.

Looking at code, it was hard for me to understand the meaning of `target`, `targetNode` and their comparison later on. If I'm understanding the issue correctly, scrolling in UI causes `event.target` to be `ul`, while navigation with arrows alternates between `ul` and child `span`. I have to admit, this is very surprising to me. I would expect that [the 'scroll' event](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/DynamicInlineScroll.es.js#L36) always has `ul` as target.

_If_ we need to make a distinction between different behaviors of types of scrolling, let's leave an explicit marker in code that we are doing this. Something like:
```js
const keyboardNavigation = listElement.nodeName !== target.nodeName
```

Reply:
This is happening as I explained [here](https://github.com/liferay-frontend/liferay-portal/pull/2838#discussion_r1023086400), only using JAWS and navigating with UP and DOWN keys on the list items.